src/soho/tag/soho-tag.component.ts

```diff
@@ -0,0 +1,200 @@
+})
+export class SohoTagComponent implements AfterViewInit, OnDestroy {
+ // -------------------------------------------
+  // Supported tag types.
+  // -------------------------------------------
+
+  static ALERT: SohoTagType = 'alert';
+  static GOOD: SohoTagType = 'good';
+  static SECONDARY: SohoTagType = 'secondary';
+  static ERROR: SohoTagType = 'error';
+  static DEFAULT: SohoTagType = undefined;
+
+  // -------------------------------------------
+  // Component Output
+  // -------------------------------------------
+
+  /**
+   * Fired before a tag item is removed.
+   */
+  @Output() beforeRemove = new EventEmitter<SohoTagBeforeRemoveEvent>();
+
+  /**
+   * Fired after a tag item is removed.
+   */
+  @Output() click = new EventEmitter<SohoTagAfterRemoveEvent>();
+
+  /** Options. */
+  private options: SohoTagOptions = {};
+
+  @HostBinding('class.tag') get isTag() {
+    return true;
+  }
+
+  @HostBinding('class.alert')
+  get alert() {
+    return this.tagType === SohoTagComponent.ALERT;
+  }
+
+  @HostBinding('class.good')
+  get good() {
+    return this.tagType === SohoTagComponent.GOOD;
+  }
+
+  @HostBinding('class.secondary')
+  get secondary() {
+    return this.tagType === SohoTagComponent.SECONDARY;
+  }
+
+  @HostBinding('class.error')
+  get error() {
+    return this.tagType === SohoTagComponent.ERROR;
+  }
+
+  @HostBinding('class.is-clickable') @Input() isClickable: boolean;
+
+  @HostBinding('class.is-dismissible') @Input() isDismissible: boolean;
+
+  @Input('soho-tag') set sohoTag(type: SohoTagType) {
+    this.tagType = type ? type : SohoTagComponent.DEFAULT;
+  }
+
+  private tagType: SohoTagType;
+
+  private jQueryElement: JQuery;
+
+  private tag: SohoTag;
+
+  /**
+   * Creates an instance of SohoTagComponent.
+   *
+   * @param {ElementRef} element
+   * @param {NgZone} ngZone
+   * @memberof SohoTagComponent
+   */
+  constructor(
+    private element: ElementRef,
+    private ngZone: NgZone) { }
+
+  ngAfterViewInit() {
+    // call outside the angular zone so change detection
+    // isn't triggered by the soho component.
+    this.ngZone.runOutsideAngular(() => {
+      // assign element to local variable
+      this.jQueryElement = jQuery(this.element.nativeElement);
+
+      // initialise the tag control
+      this.jQueryElement.tag(this.options);
+
+      // extract the api
+      this.tag = this.jQueryElement.data('tag');
+
+      // @todo - add event binding control so we don't bind if not required.
+      // this.jQueryElement
+
+      this.jQueryElement
+        .on('beforetagremove', (e: JQuery.Event, element: HTMLElement) => this.onBeforeTagRemove(e, element) )
+        .on('click', (e: JQuery.Event) => this.onClick(e) );
+    });
+  }
+
+  private onBeforeTagRemove(e: JQuery.Event, element: HTMLElement) {
+    this.ngZone.run(() => this.beforeRemove.next(e) );
+  }
+
+  private onClick(e: JQuery.Event) {
+    this.ngZone.run(() => this.click.next(e) );
```

Comment:
Depending on when these are executed they may not cause change detection.
For example in soho-tabs we have the events hooked up before the initialization of the tabs control so that we get the initial "tabs activated event". 

In that case we needed to add a setTimeout to the ngZone.run() so change detection would fire.
```
this.ngZone.run(() => setTimeout(() => this.tabActivated.next(e), 1));
```

If that's not the case then this should be fine as user initiated events will initiate change detection. The only other thing that could be an issue is if events are fired from soho w/o any user event being the cause.